threads/cpuinfo.h

```diff
@@ -135,6 +141,7 @@ unsigned int CPUInfo::getThreadId() {
   // It looks like thread id's are always multiples of 4, so...
   return GetCurrentThreadId() >> 2;
 #elif defined(__APPLE__)
+  // FIXME: is this page size dependent?
```

Comment:
I don't _think_ thread ID logic is different in OS X on Silicon but you should check.

Reply:
Hmm, I wrote a test program (similar to above), where I created a bunch of consecutive threads, and there is a difference in the offset between each thread ID on ARM vs x86. But also I didn't find a 4K offset in the IDs on x86 either.

On x86 (actually Rosetta on ARM), I got an offset of 536576 (0x83000) between each thread, while on ARM, I got 573440 (0x8C000). Am I doing something wrong, or is the offset not in fact 4K?

```c
#include <pthread.h>
#include <stdio.h>

void* dummy(void* args) { return NULL; }

int main() {
    pthread_t last = NULL;
    pthread_t cur;
    for (int i = 0; i < 100; i++) {
        pthread_create(&cur, NULL, dummy, NULL);
        size_t last_i = (size_t) last;
        size_t cur_i = (size_t) cur;
        printf("%lu - %lu = %lu\n", cur_i, last_i, cur_i - last_i);
        last = cur;
    }
}
```

Reply:
Note that the first one is a multiple of 4K, while the other is a multiple of 16K, which was what I was wondering about.

Reply:
Ah, I see. So should I change this to divide off by 16K instead of 4K for Apple Silicon? I'm not sure I fully understand why we divide off rather than just use the raw PID.

Reply:
The reason is that we want the thread IDs to be "compact" and likely to be effectively hashable.